### PR TITLE
Detox/E2E: Messaging (at-mentions, channel mentions, autocomplete etc) e2e tests in Gekidou

### PR DESCRIPTION
**Pull request by josephbaylon**
#### Summary
- Added e2e for messaging at-mention, channel mention, permalink, channel link, and associated messaging smoke test
- Added e2e for autocomplete at-mention, channel mention, emoji suggestion, slash suggestion, post draft, edit post, edit channel header, edit channel, and associated autocomplete smoke test
- Added/fixed some testIDs
- Created associated zephyr test cases under `Mobile V2` folder

Mobile V2/Autocomplete:
[MM-T4878](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4878) - At-Mention
[MM-T4879](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4879) - Channel Mention
[MM-T4880](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4880) - Emoji Suggestion
[MM-T4881](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4881) - Slash Suggestion
[MM-T4882](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4882) - Post Draft
[MM-T4883](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4883) - Edit Post
[MM-T4884](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4884) - Edit Channel Header
[MM-T4885](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4885) - Edit Channel

Mobile V2/Messaging:
[MM-T4874](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4874) - At-Mention
[MM-T4875](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4875) - Channel Mention
[MM-T4876](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4876) - Permalink
[MM-T4877](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4877) - Channel Link

Mobile V2/Smoke Test:
[MM-T4886](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4886) - Autocomplete
[MM-T4786_5](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4786) - Messaging - should be able to post a message with at-mention and channel mention
[MM-T4786_6](https://mattermost.atlassian.net/projects/MM?selectedItem=com.atlassian.plugins.atlassian-connect-plugin:com.kanoah.test-manager__main-project-page#!/testCase/MM-T4786) - Messaging - should be able to post labeled permalink and labeled channel link

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43976

#### Screenshots
Screen Shot 2022-06-24 at 11 01 42 AM

Failed tests due to known issues:
https://mattermost.atlassian.net/browse/MM-45279
https://mattermost.atlassian.net/browse/MM-45217

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by enahum** [comment]
Adding another reviewer so that they can get familiar with e2e tests

File: detox/e2e/test/autocomplete/edit_channel.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    });
+
+    afterAll(async () => {
+        // # Log out
+        await CreateOrEditChannelScreen.back();
+        await ChannelInfoScreen.close();
+        await ChannelScreen.back();
+        await HomeScreen.logout();
+    });
+
```
Comment (larkox) [inline detox/e2e/test/autocomplete/edit_channel.e2e.ts:59]: Should we add tests so it doesn't show autocomplete on other inputs?

File: detox/e2e/test/autocomplete/edit_post.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        await EditPostScreen.messageInput.clearText();
+    });
+
+    afterAll(async () => {
+        // # Log out
+        await EditPostScreen.close();
+        await ChannelScreen.back();
+        await HomeScreen.logout();
+    });
+
```
Comment (larkox) [inline detox/e2e/test/autocomplete/edit_post.e2e.ts:64]: For this and others... should we make sure the autocomplete component is completely inside the device screen? (Not sure if supported by this)

File: detox/e2e/test/autocomplete/emoji_suggestion.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import {expect} from 'detox';
+
+describe('Autocomplete - Emoji Suggestion', () => {
+    const serverOneDisplayName = 'Server 1';
+    const channelsCategory = 'channels';
+    const emojiName = 'fox_face';
+    const emojiNameFirst2Chars = emojiName.substring(0, 2);
+    const emojiName3rdToLastChars = emojiName.substring(2);
+    let emojiSuggestionAutocomplete: any;
+
```
Comment (larkox) [inline detox/e2e/test/autocomplete/emoji_suggestion.e2e.ts:33]: Most of these tests are suggesting at the beginning of the input. Should we have tests that add some "lorem ipsum" at the beginning of the input and then the suggestion?

**Review by larkox — APPROVED** [review]
Thanks for the wait! LGTM, just a few comments.

File: detox/e2e/test/autocomplete/emoji_suggestion.e2e.ts (hunk 3 above)
Comment (josephbaylon) [reply detox/e2e/test/autocomplete/emoji_suggestion.e2e.ts:33]: That's covered in at_mention.e2e.ts

**Pull request merged**